marconi-sidechain/src/Marconi/Sidechain/Api/Query/Indexers/MintBurn.hs

```diff
@@ -63,7 +63,9 @@ queryByPolicyAndAssetId env policyId assetId slotNo = do
       tryReadTMVar $
         env ^. sidechainEnvIndexers . sidechainMintBurnIndexer . mintBurnIndexerEnvIndexer
```

Comment:
I didn't pay attention to it in the past. It sounds like the query will fail if the indexer is busy indexing stuff. Do we really want this?

Reply:
It's not going to fail anymore. I removed the `Left ...`, and instead made it  return a `Right []` (empty result).

Reply:
Isn't it even worse? We return an empty result just because the indexer is "busy" (ie. the `MVar` content was taken by another thread).

Reply:
> Isn't it even worse?

Maybe yeah. I just made it consistent with the Utxo indexer. 

Do we want to use `takeTMVar` instead, which retries until it succeeds as a temp solution?

Reply:
Maybe better yes, I prefer the indexer to hang rather than returning inconsistent results.

Reply:
I'll change it then

Reply:
After changing it, it requires a bit of changes in `Marconi.ChainIndex.Indexers`. Therefore, I'll do in a separate PR.